qupulse/hardware/awgs/base.py

```diff
@@ -162,17 +166,26 @@ def __str__(self) -> str:
                " Use force to overwrite.".format(self.name)
 
 
+AllowedProgramTypes = Union[Loop,Sequence[LinSpaceNode],]
+
+class _ProgramType(Enum):
+    FSP = -1
+    Loop = 0
+    Linspace = 1
+
+
 class ProgramEntry:
     """This is a helper class for implementing awgs drivers. A driver can subclass it to help organizing sampled
     waveforms"""
-    def __init__(self, loop: Loop,
+    def __init__(self, program: AllowedProgramTypes,
```

Comment:
I think it might make sense to create a new class

Reply:
a new class for what specifically?

Reply:
I do not know if this is explicitly necessary right now - at least to ensure compatibility with other awg-drivers, one only has to rename a few 'loop's to 'program' in function calls, since ProgramEntry is only really called internally. Then one can see from there if contructing something new is necessary...